insights/parsers/nftables.py

```diff
@@ -0,0 +1,198 @@
+        \t\t}
+        \t\t
+        \t\tchain FORWARD {
+        \t\t	type filter hook forward priority -150; policy accept;
+        \t\t}
+        \t\t
+        \t\tchain OUTPUT {
+        \t\t	type route hook output priority -150; policy accept;
+        \t\t}
+        \t\t
+        \t\tchain POSTROUTING {
+        \t\t	type filter hook postrouting priority -150; policy accept;
+        \t\t	oifname "virbr0" meta l4proto udp udp dport 68 counter packets 0 bytes 0 # CHECKSUM fill
+        \t\t}
+        }
+        table ip nat {
+        \t\tchain PREROUTING {
+        \t\t	type nat hook prerouting priority -100; policy accept;
+        \t\t}
+        \t\t
+        \t\tchain INPUT {
+        \t\t	type nat hook input priority 100; policy accept;
+        \t\t}
+        \t\t
+        \t\tchain POSTROUTING {
+        \t\t	type nat hook postrouting priority 100; policy accept;
+        \t\t	ip saddr 192.168.122.0/24 ip daddr 224.0.0.0/24 counter packets 3 bytes 232 return
+        \t\t	ip saddr 192.168.122.0/24 ip daddr 255.255.255.255 counter packets 0 bytes 0 return
+        \t\t	meta l4proto tcp ip saddr 192.168.122.0/24 ip daddr != 192.168.122.0/24 counter packets 0 bytes 0 masquerade to :1024-65535
+        \t\t	meta l4proto udp ip saddr 192.168.122.0/24 ip daddr != 192.168.122.0/24 counter packets 0 bytes 0 masquerade to :1024-65535
+        \t\t	ip saddr 192.168.122.0/24 ip daddr != 192.168.122.0/24 counter packets 0 bytes 0 masquerade
+        \t\t}
+        \t\t
+        \t\tchain OUTPUT {
+        \t\t	type nat hook output priority -100; policy accept;
+        \t\t}
+        }
+
+    Raises:
+        SkipException: When content is empty.
+        ParseException: When content can not be parsed.
+
+    Examples:
+        >>> type(nftables_obj)
+        <class 'insights.parsers.nftables.NFTListRules'>
+
+    """
+
+    def parse_content(self, content):
+
+        if not content:
+            raise SkipException("No Contents")
+
+        self.data = {}
+        nst_cnt = 0
+        idx_key = ''
+        idx_key_2 = ''
+        for line in get_active_lines(content):
+            if line.endswith('{') and nst_cnt == 0:
+                idx_key = line.split('{')[0].strip()
+                self.data[idx_key] = {}
+                nst_cnt += 1
+            elif not ((line.endswith('{')) or (line.endswith('}'))) and\
+                    idx_key and idx_key_2 and nst_cnt == 2:
+                self.data[idx_key][idx_key_2].append(line)
+            elif line.endswith('{') and nst_cnt == 1 and idx_key:
+                idx_key_2 = line.split('{')[0].strip()
+                self.data[idx_key][idx_key_2] = []
+                nst_cnt += 1
+            elif line.endswith('}') and nst_cnt == 2 and idx_key_2:
+                idx_key_2 = ''
+                nst_cnt -= 1
+            elif line.endswith('}') and nst_cnt == 1 and idx_key:
+                idx_key = ''
+                nst_cnt -= 1
```

Comment:
@vishwanathjadhav this may just be my lack of knowledge about `nft` but could you just look for `startswith` table | chain to determine the level here?  What about using `self.tables` instead of `self.data` which is not very descriptive, to represent the type of data actually being stored?